docs/source/notes/numerical_accuracy.rst

```diff
@@ -110,6 +110,13 @@ reduced-precision reductions are problematic, they can be turned off with
 
 For more information see :ref:`allow_fp16_reduced_precision_reduction<fp16reducedprecision>` and :ref:`allow_bf16_reduced_precision_reduction<bf16reducedprecision>`
 
+Reduced Precision Reduction for FP16 and BF16 in Scaled Dot Product Attention (SDPA)
+------------------------------------------------------------------------------------
+The SDPA math backend, when using FP16/BF16 inputs, exhibits significant numerical errors due to the low-precision intermediate buffers. To mitigate this issue, the default behavior now involves converting FP16/BF16 inputs to FP32. Computations are performed in FP32/TF32, and the final FP32 results are then converted back to FP16/BF16. This will improved numerical accuracy for math backend with FP16/BF16 inputs, but potentially caused the performance regression in the math backend as computations shift from FP16/BF16 BMM to FP32/TF32 BMM/Matmul.
```

Comment:
some doc nits:
```suggestion
A naive SDPA math backend, when using FP16/BF16 inputs, can accumulate significant numerical errors due to the usage of low-precision intermediate buffers. To mitigate this issue, the default behavior now involves upcasting FP16/BF16 inputs to FP32. Computations are performed in FP32/TF32, and the final FP32 results are then downcasted back to FP16/BF16. This will improve numerical accuracy of the final output for the math backend with FP16/BF16 inputs, but increases memory usage and may cause performance regressions in the math backend as computations shift from FP16/BF16 BMM to FP32/TF32 BMM/Matmul.
```